core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1976,6 +1976,56 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  /**
+   * Returns a qualified path object.
+   * @param defaultUri default uri corresponding to the filesystem provided.
+   * @param workingDir the working directory for the particular child path wd-relative names.
+   * @param path Path instance based on the path string specified by the user.
+   * @return Returns a qualified path object
+   */
+  def makeQualified(defaultUri: URI, workingDir: Path, path: Path): Path = {
```

Comment:
If this is 'ported' from a Hadoop API method, it's good to note that inside the method as a comment, at least.

Reply:
Make this method `private` -- can be right?